pkg/agent/controller/bgp/controller_test.go

```diff
@@ -850,6 +951,58 @@ func TestBGPPolicyUpdate(t *testing.T) {
 					ipv4Peer2Config,
 					ipv6Peer2Config,
 				},
+				&confederationConfig{101, sets.New[uint32](uint32(65001))},
+			),
+			expectedCalls: func(mockBGPServer *bgptest.MockInterfaceMockRecorder) {
+				mockBGPServer.Start(gomock.Any())
+				mockBGPServer.Stop(gomock.Any())
```

Comment:
maybe you should have `mockBGPServer.Stop(gomock.Any())` at the beginning, and `mockBGPServer.Start(gomock.Any())` at the end, for clarity

Reply:
You are right. That's clearer, after all,  `mockBGPServer.Stop(gomock.Any())` is called before `mockBGPServer.Start(gomock.Any())` in this test.